object-store/src/main/java/org/hypertrace/config/objectstore/IdentifiedObjectStore.java

```diff
@@ -205,6 +198,29 @@ public List<ContextualConfigObject<T>> upsertObjects(RequestContext context, Lis
         .collect(Collectors.toUnmodifiableList());
   }
 
+  public List<ContextualConfigObject<T>> deleteObjects(
+      RequestContext requestContext, List<String> contexts) {
+    List<ConfigToDelete> configsToDelete =
+        contexts.stream()
+            .map(
+                context ->
+                    ConfigToDelete.newBuilder()
+                        .setResourceName(this.resourceName)
+                        .setResourceNamespace(this.resourceNamespace)
+                        .setContext(context)
+                        .build())
+            .collect(Collectors.toUnmodifiableList());
+    return requestContext
+        .call(
+            () ->
+                this.configServiceBlockingStub.deleteConfigs(
+                    DeleteConfigsRequest.newBuilder().addAllConfigs(configsToDelete).build()))
+        .getDeletedConfigsList()
+        .stream()
+        .map(deletedConfig -> processDeleteResult(requestContext, deletedConfig))
```

Comment:
If I delete 10 things and the first fails to process, no notifications will be sent for any of them. We should probably wait to throw until the end - if at all.

Reply:
Right, okay. Also should we throw an error or just log one and continue?🤔 What do you suggest?

Reply:
So thinking through it, what's likely to fail in this function? The conversion from value to proto right? If that happens, from a delete API perspective, it's not _really_ an error - it deleted just fine (I'd call this an issue with existing single delete too).

So I would probably process a result by gathering the ID that was deleted and an optional config if we can parse it. 

On the method return (both singular and bulk), gather all the deleted IDs regardless of whether we parsed the configs or not and return those. Basically method should only error if it actually didn't delete.

On the config change event - on the read side, does it support leaving the prev value unset and just setting the ID? If so, seems better than nothing. If not, skip sending it. Either way, log a warning for that, but still return the id to the delete result. 

If

Reply:
Skipping sending it for now.
We probably can set the prev value to some null value, but would need to send the config type as part of the delete notification which we can't parse unless we can successfully convert the value to proto. I'm not sure if there would be any use if we send the delete notification setting the config type to say an empty string. Also, not sure if they would get parsed with the current consumer side implementation. What do you think? 🤔

Reply:
(so yes, agree to skip sending for now)